cmd/copy.go

```diff
@@ -217,18 +218,50 @@ func (raw rawCopyCmdArgs) cook() (cookedCopyCmdArgs, error) {
 	}
 
 	// Note: new implementation of list-of-files only works for remove command and upload for now.
-	if cooked.fromTo == common.EFromTo.BlobTrash() || cooked.fromTo == common.EFromTo.FileTrash() || cooked.fromTo.From() == common.ELocation.Local() {
-		cooked.listOfFilesLocation = raw.listOfFilesToCopy
-	} else if cooked.fromTo == common.EFromTo.BlobFSTrash() {
-		// Note: when the ADLS Gen2 interop happens eventually, we should keep the current blob implementation
-		// so that users can still use include/exclude flags,
-		// and we can shift to the new directory delete API (equivalent to the current bfs implementation)
-		// if no include/exclude is supplied.
-		if len(raw.include) > 0 || len(raw.exclude) > 0 {
+	if fromTo.To() == common.ELocation.Unknown() || cooked.fromTo.From() == common.ELocation.Local() {
+		// This handles both list-of-files and include-path as a list enumerator.
+		// This saves us time because we know *exactly* what we're looking for right off the bat.
+		// Note that exclude-path is handled as a filter unlike include-path.
+
+		// to garbage or to local
+		if (len(raw.include) > 0 || len(raw.exclude) > 0) && cooked.fromTo == common.EFromTo.BlobFSTrash() {
 			return cooked, fmt.Errorf("include/exclude flags are not supported for this destination")
 		}
 
-		cooked.listOfFilesLocation = raw.listOfFilesToCopy
+		// unbuffered so this reads as we need it to rather than all at once in bulk
+		listChan := make(chan string)
+		var f *os.File
+
+		if raw.listOfFilesToCopy != "" {
+			f, err = os.Open(raw.listOfFilesToCopy)
+
+			if err != nil {
+				return cooked, fmt.Errorf("cannot open %s file passed with the list-of-file flag", raw.listOfFilesToCopy)
+			}
+		}
+
+		go func() {
+			defer close(listChan)
+
+			if f != nil {
```

Comment:
Can the user specify both list-of-files and includePathList?  Maybe we should do something in validation to prevent that, since its kind of confusing?  (It's hard to think of a case where it makes sense, and if it did make sense, I think user would expect the two criteria to be AND-ed together - only transfer things that are covered by both - whereas there's no way we want to AND them. We are OR-ing them because AND-ing is just too complicated to implement).

Reply:
Yes. That was the intent. I'll prevent the combo, I suppose.